### PR TITLE
code sync

### PR DESCRIPTION
**Pull request by mattmanning**
Elaborate on code sync, and update data persistence to reflect new volumes behavior.

# Release Playbook
- [x] Rebase against master
- [x] Code review
- [x] Merge into master
- [ ] Verify changes at http://site-staging.convox.com
- [ ] Promote release on `site-production` in the `convox/production` Rack

### CODE REVIEW

**Comment by mattmanning** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;yE9p2srw&#x2F;391-better-code-sync-docs"> Better Code Sync docs</a>

File: _development/running-locally.md [hunk 1]
```diff
@@ -40,9 +40,19 @@ To exit `convox start` press `Ctrl+C`.
...
+
+Files edited or created in the container will also get synced back into the host. This comes in handy for frameworks like Rails, for example, where lots of code is generated by the framework. You can run a generator inside the container and then see the files on your host where they can be edited, committed to version control, etc.
+
+### Why not volumes?
+
+Volumes are useful in many scenarios (see the Data persistence info below), but are not ideal for code syncing. Docker volumes use overlay mounts. That means the host directory is mounted over the corresponding container directory, making any files that previously existed exclusively on the container unavailable. This is problematic if you want to build useful files into a base Docker image that your app's Dockerfile inherits from.
+
+### Disabling code sync
+
+If you don't want to use code sync you can disable it by passing the `--sync=false` option to `convox start`.
```
Comment (ddollar) [inline _development/running-locally.md]: This should be `--no-sync`

**Comment by ddollar** [comment]
Let's also add a separate doc specific to code sync that describes details of how it works, including `dockerignore` etc

**Comment by ianmalott** [comment]
Do you think it'd be useful to still keep a brief **Code sync** section in the _Running Locally_ doc that links to the standalone _Code sync_ doc?

**Comment by mattmanning** [comment]
@ianmalott Good suggestion. Just added.

**Pull request merged**